charts/sonarqube-dce/values.yaml

```diff
@@ -105,14 +105,15 @@ searchNodes:
 
   ## We usually don't make specific resource recommendations, as they are heavily dependent on
   ## The usage of SonarQube and the surrounding infrastructure.
-  ## Adjust these values to your needs, but make sure that the memory limit is never under 4 GB
+  ## Adjust these values to your needs, but make sure that the memory limit is never under 3GB.
+  ## Those default are based on the default -Xmx2G -Xms2G settings of searchNodes sub processes.
   resources:
     limits:
       cpu: 800m
-      memory: 4096M
+      memory: 3072M
     requests:
       cpu: 400m
-      memory: 2Gi
+      memory: 3072M
```

Comment:
Why setting requests and limits to the same value here? I see that here and below, you try to set an extra Gi to the processes, what's the reason?

Reply:
So the reasoning is as follow.

For DCE chart we have full control so I suggest we go for request=limit which is much better for memory based application like java.

we are always adding an overhead of 1Gi compared to the sum of XMX and XMS cause those values are not a hard cap. They cap only some part of the JVM heap memory allocation so it is always recommended to have xmx/xms at 80% of the available memory (here materialized with request/limits).

Basically this is a strongly opinionated towards stability.

Reply:
shall we add this recommendation in the readme for those cases? It can be added in parentheses next to the values (`xx.requests` and `xx.limits`) description.